Gems/GraphCanvas/Code/Source/Components/SceneComponent.cpp

```diff
@@ -1246,23 +1245,22 @@ namespace GraphCanvas
         saveData->ClearConstructData();
         for (AZ::Entity* currentEntity : m_graphData.m_nodes)
         {
-            GraphCanvasConstructSaveData* constructSaveData = nullptr;
-
             if (GraphUtils::IsComment(currentEntity->GetId()))
             {
-                constructSaveData = aznew GraphCanvasConstructSaveData();
+                GraphCanvasConstructSaveData* constructSaveData = aznew GraphCanvasConstructSaveData();
                 constructSaveData->m_constructType = ConstructType::CommentNode;
-            }
-            else if (GraphUtils::IsNodeGroup(currentEntity->GetId()))
-            {
-                constructSaveData = aznew GraphCanvasConstructSaveData();
-                constructSaveData->m_constructType = ConstructType::NodeGroup;
+                EntitySaveDataRequestBus::Event(currentEntity->GetId(), &EntitySaveDataRequests::WriteSaveData, constructSaveData->m_saveDataContainer);
+                saveData->m_constructs.push_back(constructSaveData);
+                continue;
             }
 
-            if (constructSaveData)
+            if (GraphUtils::IsNodeGroup(currentEntity->GetId()))
             {
+                GraphCanvasConstructSaveData* constructSaveData = aznew GraphCanvasConstructSaveData();
+                constructSaveData->m_constructType = ConstructType::NodeGroup;
                 EntitySaveDataRequestBus::Event(currentEntity->GetId(), &EntitySaveDataRequests::WriteSaveData, constructSaveData->m_saveDataContainer);
                 saveData->m_constructs.push_back(constructSaveData);
+                continue;
             }
         }
```

Comment:
Are there future planned changes here? Just wondering why the refactor to duplicate the call to the ebus?

Reply:
This particular change happened organically while trying to understand how selection was handled in the first place and how all of the save data was serialized. I was originally going to try to inject the selection data here but it was not going to catch all of the cases and would impact script canvas. Thankfully, since all of the items read and write data via a bus, I was able to use the bus router to connect to all of the graph items noninvasively. I could revert this block but I think it's cleaner. It would be even better to use a lambda and pass in the container, entity ID, and construct type for all three blocks.